upgrade/src/main/java/org/odk/collect/upgrade/Upgrade.kt

```diff
@@ -0,0 +1,13 @@
+package org.odk.collect.upgrade
+
+interface Upgrade {
+
+    /**
+     * If this returns a non-`null` value, the [Upgrade] will only be run once (the next app
+     * upgrade). [Upgrade] implementations that return `null` from [Upgrade.key] will be run on
+     * every app upgrade.
+     */
+    fun key(): String?
```

Comment:
@grzesiek2010 hopefully this makes more sense now? If not, let's try and work out something we both like!

Reply:
Yes now it's a bit clearer. I'm not sure if it's the best option but probably it would be difficult to implement something simpler but with backwards compatibility at the same time. Let's leave it maybe next time you or me touch that class will be able to improve it.